lib/perl/Genome/InstrumentData/AlignmentResult/Merged.pm

```diff
@@ -219,24 +226,70 @@ sub create {
         for my $alignment (@alignments) {
             $alignment->add_user(user => $self, label => 'uses');
         }
-
-        return 1;
-    };
-    if (my $error = $@) {
-        $tx->rollback();
-        die $error;
-    } elsif ($rv ne 1) {
+    }
+    catch {
         $tx->rollback();
-        $self->error_message('Unexpected return value: ' . $rv);
-        die $self->error_message;
-    } else {
-        $tx->commit();
+        die $self->error_message('Merge failed due to error: ' . $_);
     }
+    finally {
+        for my $allocation (@temp_allocations) {
+            $allocation->delete;
+        }
+        unless ($_) {
+            $tx->commit();
+            $self->debug_message('Merge done');
+        }
+    };
 
     $self->debug_message("Resizing the disk allocation...");
     if ($self->_disk_allocation) {
-        unless ($self->_disk_allocation->reallocate) {
-            $self->warning_message("Failed to reallocate disk allocation: " . $self->_disk_allocation->id);
+        unless (eval { $self->_disk_allocation->reallocate }) {
+            $self->warning_message("Failed to reallocate my disk allocation with id (%s). Eval returned (%s) ", $self->_disk_allocation->id, $@);
+        }
+        $self->output_dir($self->_disk_allocation->absolute_path); #update if was moved
```

Comment:
It seems like this little piece of code to reallocate the disk allocation of a software result and update the output_dir if necessary belongs in SoftwareResult.pm  (It seems weird to me for child classes to be modifying output_dir at whim)